### ISSUE TITLE
macos @available regression

### ISSUE THREAD

**Issue by billzez** [ziglang/zig#13503]
### Zig Version

0.11.0-dev.116+41b7e40d7

### Steps to Reproduce and Observed Behavior

```
int main() {
  if (@available(macOS 12.0, *)) {
  }
  return 0;
}
```

```
$ zig build-exe -target x86_64-macos test.m
MachO Flush... error(link): undefined reference to symbol '___isPlatformVersionAtLeast'
error(link):   first referenced in '/home/bill/.cache/zig/o/b5f10fec5c11c1177e347bbb7949330b/test.o'
error: UndefinedSymbolReference
```

### Expected Behavior

This linked fine with in `0.10.0-dev.4476+0f0076666`.

**Comment by billzez** [comment]
If I change the code to following it works.

```
int main() {
  if (@available(macOS 11.0, *)) {
  }
  return 0;
}
```

But I need to test for 12.0.

**Comment by kubkon** [comment]
This seems to work as of Zig version `0.11.0-dev.2374+a329450aa`
<img width="491" alt="Screenshot 2023-04-05 at 08 52 24" src="https://user-images.githubusercontent.com/1519747/230003539-30e00108-462b-4fd1-b80b-51a457a4f146.png">
Could you confirm this is also the case for you with latest Zig master?
